src/UI/templates/js/Input/Field/tagInput.js

```diff
@@ -0,0 +1,123 @@
+/**
+ * Wraps the BootstrapTagsInput
+ *
+ * @author Fabian Schmid <fs@studer-raimann.ch>
+ */
+var il = il || {};
+il.UI = il.UI || {};
+il.UI.Input = il.UI.Input || {};
+(function ($) {
+    il.UI.Input.tagInput = (function ($) {
+
+
+        /**
+         *
+         * @param raw_id
+         * @param config
+         */
+        var init = function (raw_id, config) {
+            var _DEBUG = false;
+            var _CONFIG = {
+                id: '',
+                options: [],
+                selected_options: [],
+                extendable: false,
+                suggestion_starts: 1,
+                max_chars: 2000,
+                suggestion_limit: 50,
+                debug: _DEBUG,
+                allow_duplicates: false,
+                highlight: true,
+                hint: true,
+                tag_class: "label label-primary"
+            };
+
+            var _ELEMENTS = {
+                hidden_template: null,
+                container: null
+            };
+
+            var _log = function (key, data) {
+                if (!_DEBUG) {
+                    return;
+                }
+                console.log("***********************");
+                console.log(key + ":");
+                console.log(data);
+            };
+
+
+            var _initBloodhound = function () {
+                var bloodHoundObj = new Bloodhound({
+                    datumTokenizer: Bloodhound.tokenizers.whitespace,
+                    queryTokenizer: Bloodhound.tokenizers.whitespace,
+                    local: _CONFIG.options
+                });
+                bloodHoundObj.initialize();
+
+                return bloodHoundObj;
+            };
+
+            // Initialize ID and Configuration
+            var id = '#' + raw_id;
+            _CONFIG = $.extend(
+                _CONFIG
+                , config);
```

Comment:
Here you override the above defined defaults, with the ones defined in PHP. I would rather check if configs contains all needed default instead of just using js defaults if they are not set by configs. This pattern might produce errors which are hard to track. Default set by php is not used, since not set correctly, instead js default set by _CONFIG is used...